app/configurator/components/chart-controls/color-ramp.tsx

```diff
@@ -1,34 +1,209 @@
+
     if (canvas && context) {
-      canvas.style.width = `${width}px`;
-      canvas.style.height = `${height}px`;
+      context.clearRect(0, 0, width, height);
       canvas.style.imageRendering = "-moz-crisp-edges";
       canvas.style.imageRendering = "pixelated";
-      context.clearRect(0, 0, width, height);
-      for (let i = 0; i < nbClass; ++i) {
-        context.fillStyle = colorInterpolator(i / (nbClass - 1));
-        context.fillRect((width / nbClass) * i, 0, width / nbClass, height);
+      canvas.style.borderRadius = "2px";
+
+      const [widthPerClass, numberOfSteps] =
+        nbClass > width ? [1, width] : [width / nbClass, nbClass];
+
+      for (let i = 0; i < numberOfSteps; ++i) {
+        context.fillStyle = colorInterpolator(i / (numberOfSteps - 1));
+        context.fillRect(widthPerClass * i, 0, widthPerClass, height);
       }
     }
+  }, [colorInterpolator, height, width, nbClass]);
+
+  return <canvas ref={canvasRef} width={width} height={height} />;
+};
+
+type ColorRampFieldProps = Omit<ColorRampProps, "colorInterpolator"> & {
+  field: string;
+  path: string;
+};
+
+export const ColorRampField = ({
+  field,
+  path,
+  nbClass,
+}: ColorRampFieldProps) => {
+  const [state, dispatch] = useConfiguratorState();
+
+  const palettes = useMemo(
+    () => [...divergingPalettes, ...sequentialPalettes],
+    []
+  );
+
+  const currentPaletteName = get(
+    state,
+    `chartConfig.fields.${field}.${path}`
+  ) as DivergingPaletteType | SequentialPaletteType;
+
+  const currentPalette =
+    palettes.find((d) => d.value === currentPaletteName) || palettes[0];
+
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    highlightedIndex,
+    getItemProps,
+  } = useSelect({
+    items: palettes,
+    defaultSelectedItem: palettes.find((d) => d.value === "oranges"),
+    onSelectedItemChange: ({ selectedItem }) => {
+      if (selectedItem) {
+        dispatch({
+          type: "CHART_OPTION_CHANGED",
+          value: {
+            field,
+            path,
+            value: selectedItem.value,
+          },
+        });
+      }
+    },
   });
 
-  return <canvas ref={canvasRef} />;
+  return (
+    <Box pb={2}>
+      <Label smaller {...getLabelProps()}>
+        <Trans id="controls.color.palette">Color palette</Trans>
+      </Label>
+      <Button
+        {...getToggleButtonProps()}
+        sx={{
+          color: "monochrome700",
+          display: "flex",
+          justifyContent: "space-between",
+          alignItems: "center",
+          bg: "monochrome100",
+          p: 1,
+          height: "40px",
+          borderWidth: "1px",
+          borderStyle: "solid",
+          borderColor: "monochrome500",
+          ":hover": {
+            bg: "monochrome100",
+          },
+          ":active": {
+            bg: "monochrome100",
+          },
+          ":disabled": {
+            cursor: "initial",
+            bg: "muted",
```

Comment:
Could we add it to the button variants in the theme or is it really specific to this place ?

Reply:
Hmmm, I based this on styles from `ColorPalette`, I guess it's common for custom select elements like these in color palettes. You mean extracting the whole CSS as a button variant?

Reply:
Yes. It is not particularly problematic if it is really needed, but I think that if we have the concept of "custom buttons for color/palette pickers" maybe it should be a button variant, it would be clearer.